### PR TITLE
IEN-806 - create s3 bucket to deploy api lambda

### PR DESCRIPTION
**Pull request by coryatwater-FW**

### CODE REVIEW

File: terraform/api.tf [hunk 1]
```diff
@@ -1,14 +1,55 @@
...
+        }
+      }
+    ]
+  })
+}
+
+resource "aws_s3_bucket_object" "api_lambda" {
+  bucket = aws_s3_bucket.api.bucket
+  key    = "api-lambda-s3"
+  source = "./build/empty_lambda.zip"
```
Comment (sunghwan-park-fw) [inline terraform/api.tf:37]: How about specifying api.zip, supposedly built just before running terraform?

File: Makefile [hunk 2]
```diff
@@ -297,6 +299,7 @@ runs:
 
 sync-app:
 	aws s3 sync ./terraform/build/app s3://$(APP_SRC_BUCKET) --delete
+	aws s3 sync ./terraform/build/api s3://$(API_SRC_BUCKET) --delete
```
Comment (sunghwan-park-fw) [inline Makefile:302]: The task is `sync-app` and it shouldn't delete API.
Unlike app, api.zip is just one file so I doubt we need to delete it.
Anyway, we need to define how to upload it. We've done it by running `aws lambda update-function-code`. Now we might need two steps, first upload to s3 and update the function from s3.

File: terraform/api.tf (hunk 1 above)
Comment (coryatwater-FW) [reply terraform/api.tf:37]: ah I tried that, says error file not found, troubleshooting that now

File: terraform/iam.tf [hunk 3]
```diff
@@ -17,6 +17,31 @@ resource "aws_iam_role" "lambda" {
   })
 }
 
+resource "aws_s3_bucket_policy" "api_lambda" {
```
Comment (sunghwan-park-fw) [inline terraform/iam.tf:20]: With aws_iam_role_policy_attachment.lambda_s3, do we still need this?

File: Makefile [hunk 4]
```diff
@@ -298,16 +300,20 @@ runs:
 sync-app:
 	aws s3 sync ./terraform/build/app s3://$(APP_SRC_BUCKET) --delete
 
+upload-sync-api:
+	aws s3 sync ./terraform/build/api s3://$(API_SRC_BUCKET)
```
Comment (sunghwan-park-fw) [inline Makefile:304]: Do we need this?

**Review by dbayly-freshworks — APPROVED** [review]

**Pull request merged**